src/modules/m_joinflood.cpp

```diff
@@ -177,7 +185,12 @@ class ModuleJoinFlood : public Module
 			{
 				f->clear();
 				f->lock();
-				memb->chan->WriteNotice(InspIRCd::Format("This channel has been closed to new users for %u seconds because there have been more than %d joins in %d seconds.", duration, f->joins, f->secs));
+				if (notify) {
+					std::string message = InspIRCd::Format("This channel has been closed to new users for %u seconds because there have been more than %d joins in %d seconds.", duration, f->joins, f->secs);
```

Comment:
I believe Channel::WriteNotice() allows for an optional status character.

Reply:
You're right about the optional parameter. Changed to simplify the code.